thefuck/shells/generic.py

```diff
@@ -121,6 +122,33 @@ def put_to_history(self, command):
 
         """
 
+    def can_edit(self):
+        return False
+
+    def edit_command(self, command):
+        """Spawn default editor (or `vi` if not set) and edit command in a buffer"""
+        # Create a temporary file and write some default text
+        # mktemp somewhere
+
+        editor = os.getenv("EDITOR", "vi")
+
+        tf = tempfile.NamedTemporaryFile(
```

Comment:
NamedTemporaryFile is a context manager, so you can use it with `with`, meaning you don't have to explicitly call close or unlink it.